cli/test/lib/cypress_spec.js

```diff
@@ -172,4 +172,30 @@ describe('cypress', function () {
       })
     })
   })
+
+  context('cli', function () {
+    describe('.parseRunArguments', function () {
+      it('parses CLI cypress run arguments', async () => {
+        const args = 'cypress run --browser chrome --spec "my/test/spec.js"'.split(' ')
+        const options = await cypress.cli.parseRunArguments(args)
+
+        // TODO remove quotes around specs and other string arguments
+        expect(options).to.deep.equal({
+          browser: 'chrome',
+          spec: '"my/test/spec.js"',
+        })
+      })
+
+      it('parses CLI cypress run shorthand arguments', async () => {
+        const args = 'cypress run -b firefox -p 5005'.split(' ')
+        const options = await cypress.cli.parseRunArguments(args)
+
+        // ? should we cast the arguments to proper types
```

Comment:
For the user? We should probably have types for the arguments we echo back.

Reply:
well, that's precisely the point. If we parse it and it is ready to go into the `cypress.run` then we should try to cast

Reply:
I have added casting of `port` and `configFile` and corresponding tests, so we are good here